src/clr-angular/utils/drag-and-drop/draggable-ghost.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {animate, style, transition, trigger} from "@angular/animations";
+import {Component, ElementRef, HostBinding, NgZone, OnDestroy, Optional, Renderer2} from "@angular/core";
+import {Subscription} from "rxjs/Subscription";
+
+import {ClrDragEvent} from "./interfaces/drag-event";
+import {ClrDragEventListener} from "./providers/drag-event-listener";
+import {ClrDraggableStateRegistrar} from "./providers/draggable-state-registrar";
+
+const ANIMATION_DURATION = "0.2s ease-in-out";
+@Component({
+    selector: "clr-draggable-ghost",
+    template: `<ng-content></ng-content>`,
+    host: {class: "draggable-ghost"},
+    animations: [trigger(
+        "leaveAnimation",
+        [transition(
+            ":leave",
+            [style({left: "*", top: "*"}), animate(ANIMATION_DURATION, style({top: "{{top}}", left: "{{left}}"}))])])]
+})
+export class ClrDraggableGhost<T> implements OnDestroy {
+    private draggableGhostEl: Node;
+
+    private subscriptions: Subscription[] = [];
+
+    private initPosition: {pageX: number; pageY: number};
+    private dragPosition: {pageX: number; pageY: number};
+    private initDragDelta = {top: 0, left: 0};
+
+    @HostBinding("@leaveAnimation") leaveAnimConfig = {value: 0, params: {top: "0px", left: "0px"}};
+
+    constructor(private el: ElementRef, @Optional() private dragEventListener: ClrDragEventListener<T>,
+                private renderer: Renderer2, private ngZone: NgZone,
+                private draggableStateRegistrar: ClrDraggableStateRegistrar<T>) {
+        if (!this.dragEventListener) {
+            throw new Error("The clr-draggable-ghost component can only be used inside of a clrDraggable directive.");
+        }
+
+        this.draggableGhostEl = this.el.nativeElement;
+        this.renderer.addClass(document.body, "in-drag");
```

Comment:
Nitpick: I feel like this doesn't belong here. Maybe in the draggable itself or in the structural directive? I'd almost want this to be in a service, since it's here to make sure the app itself has the right behavior (no selection) more than having the draggable look good.

Please comment here or talk about it with the team on Slack before changing this, I'm not sure where the best place for this line of code is and I don't want you to have to change it 5 times.

Reply:
I pondered this too. But at the same, the purpose of applying this `"in-drag"` class' styles is to make the underlaying document possible to have the hovering draggable-ghost element over. It's facilitating draggable-ghost. So as I didn't see strong, compelling reasons to view this line of code outrageously out of place, I kept it here for now.

Reply:
Just a note here, we do a similar kind of thing for Responsive Navigation or Modals (not sure which one) with the `ScrollingService`.

Reply:
@Shijir did you get a chance to look at the `ScrollingService` and check if we can do something like that here?

Reply:
Looks promising. I can incorporate something similar to that.